File: 2021/05/20/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        l = []
        def dfs(root, i):
            if root is None:
                return
            
            if len(l) < i + 1:
                l.append([])
            
            l[i].append(root.val)
            
            dfs(root.left, i+1)
            dfs(root.right, i+1)
            
        dfs(root, 0)
        
        return l
        

File: 2021/05/20/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
	var l [][]int
	var dfs func(*TreeNode, int)

	dfs = func(root *TreeNode, i int) {
		if root == nil {
			return
		}

		if len(l) < i+1 {
			l = append(l, []int{})
		}

		l[i] = append(l[i], root.Val)

		dfs(root.Left, i+1)
		dfs(root.Right, i+1)
	}

	dfs(root, 0)

	return l
}
